LZA-66: ADR 2: Trust GitHub OIDC for platform development (#18)
UKHomeOffice/core-cloud: documentation/adr/001-trust-github-oidc.md

---
layout: page.njk
status: Proposed
date: git Created
tags:
  - adr
order: 1
title: 001 - Trust GitHub through OpenID Connect for platform development 
---

**Date: {{ page.date.toISOString().substring(0,10) }}**

**Status: {{ status }}**

## Context

> We need to identify a tool for our developers to maintain our infrastructure and tooling code.

Reviewer: ```suggestion
We need to identify a platform for our developers to maintain our infrastructure and tooling code.
```

GitHub Enterprise is the Home Office's strategic developer platform.

Other platforms such as GitLab, Code Commit and Bitbucket are also used in the Home Office.

> One of our principles is 'Strategic First', which means we should prefer the Home Office strategic solution.

Author: ```suggestion
One of our principles is 'Strategic First', which means we should defer to the Home Office strategic solution.
```

This ADR is only concerned with our platform team and is not a decision or limitation for customer teams.

OpenID Connect (OIDC) is an open authentication protocol that works on top of the OAuth 2.0 framework that is supported by AWS and GitHub.

There needs to be a process where trust can be revoked at short notice in case of any emergent security concerns.

## Decision

We will use GitHub to maintain our infrastructure and tooling code.

> We will use OpenId Connect to establish a trusted relationship.

Author: ```suggestion
We will use OpenId Connect to establish a trusted relationship.

We will set up one OIDC in our management account and scope IAM roles to our code repositories
```

## Consequences

> There is a trust relationship between GitHub and AWS for use by the platform team. 

Author: ```suggestion
There is a trust relationship between GitHub and AWS for use by the platform team. 

Because this is for the platform team use only, additional IDPs would need to be created for customer use, which is out of scope of this ADR.
```
